src/handlers/diff_header.rs

```diff
@@ -76,20 +67,48 @@ impl<'a> StateMachine<'a> {
                 &self.line,
                 &self.raw_line,
                 &mut self.painter,
+                &mut self.mode_info,
                 self.config,
             )?;
             handled_line = true;
         }
         Ok(handled_line)
     }
 
+    pub fn handle_diff_header_mode_line(&mut self) -> std::io::Result<bool> {
```

Comment:
(I think we should add brief doc comments to all public functions/methods although I don't think I've been great about doing that everywhere. I've been using `///` and `//` inconsistently, I don't think that matters too much until we make a library out of delta and publish the documentation)

Reply:
I tossed in a few /// comments for this & the plus & minus routines.